request.go

```diff
@@ -25,6 +25,7 @@ import (
 // Request provides methods to incrementally build http.Request object,
 // send it, and receive response.
 type Request struct {
+	context    *Context
```

Comment:
Do we need Request.context if we already have Request.chain.context?

Reply:
hmmmmmmmm probably. will check